provisioning/wildfly_channel_in_zips.adoc

```diff
@@ -0,0 +1,210 @@
+
+* https://issues.redhat.com/browse/WFLY-19130[WFLY-19130] - [Community] Publish Wildfly channel and manifest during project build
+** http://docs.wildfly.org/wildfly-proposals/build/WFLY-19130_publish_Wildfly_channel_manifest.html[Analysis Document]
+
+=== Stability Level
+
+* [x] Experimental
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+* mailto:bstansbe@redhat.com[Brian Stansberry]
+* mailto:bspyrkos@redhat.com[Bartosz Spyrko-Smietanko]
+
+=== Testing By
+
+* [x] Engineering
+
+=== Affected Projects or Components
+
+* https://github.com/wildfly/wildfly[WildFly]
+** WildFly distributions will incorporate channel metadata used to create the distributions and tooling to update the installation to a more recent version.
+* https://github.com/wildfly-extras/prospero[Prospero]
+** Prospero will be used as the library to update WildFly installations but will not surface to the user (the `installation-manager` script is the user entry point)
+* https://github.com/wildfly/wildfly-maven-plugin[wildfly-maven-plugin]
+** The WildFly Maven Plug-in provides the functionality to provision WildFly to create the distributions with the expected channel metadata.
+
+=== Other Interested Projects
+
+=== Relevant Installation Types
+
+* [x] Traditional standalone server (unzipped)
+
+== Requirements
+
+* WildFly distributions archives (zip & tgz) must contain the channel metadata that describes their installation state.
+** tracked by https://issues.redhat.com/browse/WFLY-19221[WFLY-19221]
+** Provisioned feature packs remain referenced by their Galleon feature-pack-locations (e.g. `wildfly@maven(org.jboss.universe:community-universe):current#33.0.0.Final`).
+*** Users that use Galleon tool to udpate their WildFly installation will continue to be able to do so.
+*** Galleon universe artifacts (`org.jboss.universe:community-universe` & `org.jboss.universe.producer:wildfly-producers`) will be added to the WildFly channels
+* WildFly distributions archives (zip & tgz) must contain the tooling to let users update their installations
+** tracked in https://github.com/wildfly-extras/prospero/issues/749[prospero #749]
+** Users must be able to list and select the updates to apply to their installations
+** The user MUST specify the updates to apply.
+** As this feature is experimental, the tooling should warn the user that updating their installation is an experimental mechanism
+** these operations will be using Prospero that needs to be integrated as a JBoss module in the WildFly distributions.
+* Updates must not discard any user changes to an installation (in their configuration files or JBoss modules directory)
+
+=== Non-Requirements
+
+* Changing the type of distributions during an update is not supported (in other words, it is not possible to download the zip for WildFly 33.0.0.Final and update the installation to WildFly Preview)
+* Trimming an existing installation coming from WildFly distributions with Galleon layers is not supported.
```

Comment:
This is fine but it gets me thinking about a related scenario. Is the installation-manager.sh script present in a slimmed server? If it is, what happens when the user uses it?

Reply:
If the slimming is achieved using Galleon and recorded in the `.galleon/provisioning.xml`, Prospero should only update the components in the slimmed installation. 
Or do you mean a scenario where the slimmed installation does not include Prospero modules?

Reply:
@spyrkob Both I suppose.  TBH I don't recall what I was thinking when I asked this.

Reply:
I think currently the installation-manager.sh is always provisioned even if the installation.manager package is not present, but I'm not sure if it's possible to exclude that package, maybe @yersan can confirm.